lib/api/src/entities/memory/mod.rs

```diff
@@ -0,0 +1,205 @@
+pub use shared::SharedMemory;
+use wasmer_types::{MemoryError, MemoryType, Pages};
+
+use crate::{
+    vm::{VMExtern, VMExternMemory, VMMemory},
+    AsStoreMut, AsStoreRef, ExportError, Exportable, Extern, StoreMut, StoreRef,
+};
+
+pub(crate) mod buffer;
+pub(crate) mod inner;
+pub(crate) mod location;
+pub(crate) mod shared;
+pub(crate) mod view;
+
+pub(crate) use inner::*;
+pub use view::*;
+
+/// A WebAssembly `memory` instance.
+///
+/// A memory instance is the runtime representation of a linear memory.
+/// It consists of a vector of bytes and an optional maximum size.
+///
+/// The length of the vector always is a multiple of the WebAssembly
+/// page size, which is defined to be the constant 65536 – abbreviated 64Ki.
+/// Like in a memory type, the maximum size in a memory instance is
+/// given in units of this page size.
+///
+/// A memory created by the host or in WebAssembly code will be accessible and
+/// mutable from both host and WebAssembly.
+///
+/// Spec: <https://webassembly.github.io/spec/core/exec/runtime.html#memory-instances>
+#[derive(Debug, Clone, PartialEq, Eq, derive_more::From)]
+#[cfg_attr(feature = "artifact-size", derive(loupe::MemoryUsage))]
+pub struct Memory(pub(crate) RuntimeMemory);
```

Comment:
I don't see any value for having `Memory` and `RuntimeMemory` as two separate entities (we are just calling it always underneath). Can we merge?